LayoutTests/resources/ipc.js

```diff
@@ -2,10 +2,31 @@ function randomIPCID() {
     return Math.floor(Math.random() * 10000) + 1;
 }
 
+function connectionSendWithPromisedReply(connection, destinationID, messageName, args) {
```

Comment:
I would have called this sendWithPromisedReplyForConnection or something.
It's odd / sounds unnatural to prefix the function name with "connection".